### PR TITLE
a researcher (non-project owner) should not be able to invite someone…

### LINKED ISSUE #862: a researcher (non-project owner) should not be able to invite someone to a project
Noticed it thanks to @MatthiasZepper 
written a test case (not committed anywhere at the time of writing this) which fails now. Needs its small fix.

1. Check if current user is a Researcher
2. If so, check that a project is specified
3. Get row from ProjectUsers
4. Check if the row has `owner=True`
5. If so -- user can invite another Researcher to the specified project.

### PR DESCRIPTION
**Pull request by worukan**
… to a project, written a test case to cover this
Closes #862

### CODE REVIEW

File: dds_web/api/user.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 # ENDPOINTS ############################################################################ ENDPOINTS #
 ####################################################################################################
 class AddUser(flask_restful.Resource):
-    @auth.login_required
+    @auth.login_required(role=["Super Admin", "Unit Admin", "Unit Personnel", "Project Owner"])
```
Comment (alneberg) [inline dds_web/api/user.py:46]: Frankly I think this is incorrect. I don't think "Project Owner" is a valid role as I've explained already.

Comment (worukan) [reply dds_web/api/user.py:46]: I have followed the agreed roles which are pinned down. We could remove that as well, if @inaod568 is fine with it. Someone can then update the agreement as well.

**Review by alneberg — APPROVED** [review]
Nice! Yes, including project owner as a role like this works. We need to keep this in mind for all endpoints. A quick search in the code base showed no issues with the endpoints we have currently.

**Review by inaod568 — APPROVED** [review]
Nice, works well!

**Pull request merged**